app/controllers/admin/works_controller.rb

```diff
@@ -392,4 +422,48 @@ def cancel_url
       admin_works_path
     end
     helper_method :cancel_url
+
+    def work_is_oral_history?
+      @work.genre && @work.genre.include?('Oral histories')
+    end
+    helper_method :work_is_oral_history?
+
+    # If this work is an oral history,
+    # return true if the user needs to
+    # (re)calculate combined audio derivatives.
+    # before the work can be published.
+    def need_combined_audio_derivatives?
+      # Not an oral history? Then you don't need these derivatives.
+      return false unless work_is_oral_history?
+      # Now, are they up to date?
+      existing_fingerprint = @work.oral_history_content!.combined_audio_fingerprint
+      (existing_fingerprint != CombinedAudioDerivativeCreator.new(@work).fingerprint)
+    end
+
+    def restrict_audio_member_edit?(member)
+      return false if member.work?
+      return false unless @work.published?
+      member.content_type && member.content_type.start_with?('audio')
+    end
+    helper_method :restrict_audio_member_edit?
```

Comment:
I think we talked about moving all of these helper methods out of the controller to the decorator before maybe?

Reply:
Sorry, only just now getting back to this question. We're not using decorators or presenters to wrap any of this admin-side functionality at the moment. (`app/views/admin/works/_member_list_table.html.erb` is just a garden-variety partial.)

Reply:
I'd certainly consider doing a refactor, though, now we're up to three tabs; something to consider.

Reply:
We have a sort of standard top-level Draper decorator, I think. 

As opposed to the more custom thing we call a "ViewModel" that acts as a "component", a more standard Draper decorator just "wraps" a model, usually to provide view-specific or context-specific logic -- this does get a bit confusing.  In retrospect, we maybe should have kept these two "kinds" in two different directories, if we had to have two kinds a tall. 

But what I'm thinking about is AudioWorkShowDeocrator in ./app/decorators/audio_work_show_decorator.rb.  That already has "helper" method kind of methods, keeping them out of the controller. Hypothetically, even the partials views can access these helpers, as long the as the decorated model is being passed down as an argument (instead of an undecorated model). I think sometimes we call the argument `decorator`, but sometimes just `model`, but it might really be a decorated model. There might be Reasons (tm) that it won't work for these, but I think it's worth at least a little bit of investigation maybe.

Reply:
OK (per separate discussion in Slack)`AudioWorkShowDecorator` is not being used in the admin pages.